src/Factory/ProductDocumentFactory.php

```diff
@@ -115,28 +140,28 @@ public function __construct(
      *
      * @return ProductDocument
      */
-    public function createFromSyliusSimpleProductModel(
+    public function createFromSyliusProductModel(
         ProductInterface $product,
         LocaleInterface $locale,
         ChannelInterface $channel
     ): ProductDocument {
 
-        $minProductChannelPrice = $this->getChannelPricingForChannelFromProductVariant(
-            $product->getVariants()->first(),
-            $channel
-        );
         /** @var ProductVariantInterface[] $syliusProductVariants */
         $syliusProductVariants = $product->getVariants();
 
+        /** @var ChannelPricingInterface $minProductChannelPrice */
+        $minProductChannelPrice = $this->getMinimalPriceFromVariants($syliusProductVariants, $channel);
+
         /**
          * Select minimal product price out of all variants
          */
+        $iterator = $product->getVariants()->getIterator();
+        $iterator->uasort(function (ProductVariantInterface $a, ProductVariantInterface $b) {
+            return ($a->getName() < $b->getName()) ? -1 : 1;
```

Comment:
`<=>` for the rescue! :)

Reply:
Yes, I actually just yesterday was discussing new features or 7+ and we discussed the spaceship operator and I remembered about this part